dali/operators/input/input_operator_test.cu

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <thrust/host_vector.h>
+#include <thrust/device_vector.h>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include "dali/c_api.h"
+#include "dali/pipeline/pipeline.h"
+#include "dali/test/dali_test_config.h"
+
+namespace dali::test {
+
+
+struct InputOperatorMixedTestParam {
+  int cpu_queue_depth, gpu_queue_depth;
+  bool exec_pipelined, exec_async;
+  bool cpu_input;
+};
+
+
+namespace {
+
+InputOperatorMixedTestParam input_operator_test_params_simple_executor[] = {
+        {1, 1, false, false, true},
+        {2, 2, false, false, false},
+};
+
+
+InputOperatorMixedTestParam input_operator_test_params_pipelined_executor_uniform_queue[] = {
+        {2, 2, true, false, true},
+        {3, 3, true, false, true},
+        {2, 2, true, true,  false},
+        {3, 3, true, true,  false},
```

Comment:
```suggestion
        {2, 2, true, false, true},
        {3, 3, true, false, false},
        {2, 2, true, true,  false},
        {3, 3, true, true,  true},
```
Should we make every other usage of cpu_input different for the particular executor type?

Reply:
Done